src/azul/types.py

```diff
@@ -70,3 +71,24 @@ def get_remaining_time_in_millis(self) -> int:
 
     def log(self, msg: str) -> None:
         raise NotImplementedError
+
+
+def is_optional(type_) -> bool:
```

Comment:
I like this approach in general and I came up with sth very similar for the API, before even seeing this, I swear.